tests/test_core.py

```diff
@@ -118,6 +119,12 @@ def tearDown(self):
 
     #     self.assertEqual(1, len(calls))
 
+    def test_pending_sheduler(self):
```

Comment:
This test should schedule a coro, have it finish and then see if `_async_tasks_cleanup` cleans it up. You could do that by calling `asyncio.wait` on `hass._pending_tasks` using run_coroutine_threadsafe.